homeassistant/components/netatmo/select.py

```diff
@@ -0,0 +1,163 @@
+"""Support for the Netatmo climate schedule selector."""
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+import pyatmo
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .climate import get_all_home_ids
+from .const import (
+    DATA_HANDLER,
+    DATA_SCHEDULES,
+    DOMAIN,
+    EVENT_TYPE_SCHEDULE,
+    MANUFACTURER,
+    SIGNAL_NAME,
+)
+from .data_handler import HOMEDATA_DATA_CLASS_NAME, NetatmoDataHandler
+from .netatmo_entity_base import NetatmoBase
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Netatmo energy platform schedule selector."""
+    data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
+
+    await data_handler.register_data_class(
+        HOMEDATA_DATA_CLASS_NAME, HOMEDATA_DATA_CLASS_NAME, None
+    )
+    home_data = data_handler.data.get(HOMEDATA_DATA_CLASS_NAME)
+
+    if not home_data or home_data.raw_data == {}:
+        raise PlatformNotReady
```

Comment:
It's weird to raise platform not ready if the connection handling is centralized.